File: android/src/main/kotlin/cz/appsdevteam/gptom/mappers/RegisterMapper.kt
package cz.appsdevteam.gptom.mappers

import cn.nexgo.smartconnect.model.RegisterResultV2Entity
import cz.appsdevteam.gptom.mappers.ErrorMapper
import cz.appsdevteam.gptom.core.JsonKeys

object RegisterResultMapper {
  fun toMap(e: RegisterResultV2Entity, originRef: String?): Map<String, Any?> = mapOf(
    JsonKeys.resultCode to e.resultCode,
    JsonKeys.transactionId to e.transactionId,
    JsonKeys.clientId to e.clientID,
    JsonKeys.responseMessage to e.responseMessage,
    JsonKeys.error to (e.error?.let { ErrorMapper.toMap(it) }),
    JsonKeys.originReferenceNum to originRef,
  )
}

File: ios/Classes/utils/RegisterMapper.swift
import Foundation

struct RegisterMapper {
    private init() {}

    static func toMap(
        transactionId: String,
        originReferenceNum: String?,
        clientId: String?
    ) -> [String: Any] {
        var b = JsonMapBuilder()

        b.put(JsonKeys.resultCode, 0)
        b.put(JsonKeys.transactionId, transactionId)
        b.put(JsonKeys.originReferenceNum, originReferenceNum)
        b.put(JsonKeys.clientId, clientId)
        b.put(JsonKeys.responseMessage, nil)
        b.put(JsonKeys.error, nil)

        return b.build()
    }
}
